common/src/main/java/org/mcaccess/minecraftaccess/features/point_of_interest/LockingHandler.java

```diff
@@ -207,12 +208,11 @@ private void unlock(boolean speak) {
     }
 
     private void relock() {
-        Map<String, POIGroup> entityGroups = POIEntities.getInstance().builtInGroups;
-        for (POIGroup group : entityGroups.values()) {
-            TreeMap<Double, Entity> map = group.getEntities();
-            if (!map.isEmpty()) {
-                Entity entity = map.firstEntry().getValue();
-                if (lockOnEntity(entity)) return;
+        for (POIGroup<Entity> group : POIEntities.getInstance().groups) {
+            Optional<Entity> nearest = group.getItems().stream()
+                    .min(Comparator.comparingDouble(entity -> WorldUtils.getClientPlayer().distanceTo(entity)));
+            if (nearest.map(this::lockOnEntity).orElse(false)) {
+                return;
             }
```

Comment:
Yeah, much clearer, I haven't realized that the distance info should be delayed to be produced just before where they're used.